db/warehouse/migrate/20170928185422_recreate_views_with_incorrect_primary_keys.rb

```diff
@@ -0,0 +1,247 @@
+  end
+
+  # see 20161115160857
+  def exits_up!
+    model = GrdaWarehouse::Hud::Exit
+    gh_exit_table = join_source_and_client model.arel_table
+    gh_exit_table = join_to_enrollments gh_exit_table
+    query = gh_exit_table.project(
+      *gh_exit_table.engine.column_names.map(&:to_sym).map{ |c| model.arel_table[c] },  # all the exit columns
+      enrollments_table[:id].as('enrollment_id'),                                       # a fake enrollment foreign key
+      source_client_table[:id].as('demographic_id'),                                    # a fake foreign key to the source client
+      destination_client_table[:id].as('client_id')                                     # a fake destination client foreign key
+    )
+
+    query = client_paranoia query
+    query = enrollment_paranoia query
+
+    if model.paranoid?
+      query = query.where( model.arel_table[model.paranoia_column.to_sym].eq nil )
+    end
+
+    create_view :report_exits, query
+  end
+
+  # 20161115163024
+  def health_and_dvs_up!
+    model = GrdaWarehouse::Hud::HealthAndDv
+    gh_health_and_dv_table = join_source_and_client model.arel_table
+    gh_health_and_dv_table = join_to_enrollments gh_health_and_dv_table
+    query = gh_health_and_dv_table.project(
+      *gh_health_and_dv_table.engine.column_names.map(&:to_sym).map{ |c| model.arel_table[c] },
+      enrollments_table[:id].as('enrollment_id'),
+      source_client_table[:id].as('demographic_id'),
+      destination_client_table[:id].as('client_id')
+    )
+
+    query = client_paranoia query
+    query = enrollment_paranoia query
+
+    if model.paranoid?
+      query = query.where( model.arel_table[model.paranoia_column.to_sym].eq nil )
+    end
+
+    create_view :report_health_and_dvs, query
+  end
+
+  # 20161115181519
+  def income_benefits_up!
+    model = GrdaWarehouse::Hud::IncomeBenefit
+    gh_income_benefit_table = join_source_and_client model.arel_table
+    gh_income_benefit_table = join_to_enrollments gh_income_benefit_table
+    query = gh_income_benefit_table.project(
+      *gh_income_benefit_table.engine.column_names.map(&:to_sym).map{ |c| model.arel_table[c] },
+      enrollments_table[:id].as('enrollment_id'),
+      source_client_table[:id].as('demographic_id'),
+      destination_client_table[:id].as('client_id')
+    )
+
+    query = client_paranoia query
+    query = enrollment_paranoia query
+
+    if model.paranoid?
+      query = query.where( model.arel_table[model.paranoia_column.to_sym].eq nil )
+    end
+
+    create_view :report_income_benefits, query
+  end
+
+  # 20161111214343
+  def services_up!
+    model = GrdaWarehouse::Hud::Service
+    gh_service_table = join_source_and_client model.arel_table
+    gh_service_table = join_to_enrollments gh_service_table
+    query = gh_service_table.project(
+      *gh_service_table.engine.column_names.map(&:to_sym).map{ |c| model.arel_table[c] },
+      enrollments_table[:id].as('enrollment_id'),
```

Comment:
Hmm, looks like this includes the enrolment_id, is 20161111214343 just an old migration we aren't using anymore, and this is the corrected version?

Reply:
Yeah the changes to older migrations were put in as I reasoned my way towards this migration and aren't really important.